### PR TITLE
[INLONG-5739][Manager] Add anno API in the manager client

### LINKED ISSUE #5739: [Feature][Manager] Supplement the anno API in the manager client
### Description

Supplement the anno API in the manager client.

### Use case

Supplement the anno API in the manager client.

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by haifxu**
### Prepare a Pull Request

- Fixes #5739 

### Motivation

Supplement the API that the manager web exists but the client does not exist for manager client.

### Modifications

Add anno API in the manager client.

### Verifying this change

- [x] This change added tests and can be verified as follows:

  - *org.apache.inlong.manager.client.api.inner.AnnoClientTest#testRegister*

### CODE REVIEW

**Review by leosanqing — APPROVED** [review]

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/Anno.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.manager.client.api;
+
+import org.apache.inlong.manager.pojo.user.UserRequest;
+
+public interface Anno {
```
Comment (healchow) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/Anno.java:22]: 1. Add Java doc for classes and interfaces.
2. `Anno` is unclear, suggested changing to `NoAuth`.

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/AnnoImpl.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import org.apache.inlong.manager.client.api.Anno;
+import org.apache.inlong.manager.client.api.ClientConfiguration;
+import org.apache.inlong.manager.client.api.inner.client.AnnoClient;
+import org.apache.inlong.manager.client.api.inner.client.ClientFactory;
+import org.apache.inlong.manager.client.api.util.ClientUtils;
+import org.apache.inlong.manager.common.util.Preconditions;
+import org.apache.inlong.manager.pojo.user.UserRequest;
+
+public class AnnoImpl implements Anno {
```
Comment (healchow) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/AnnoImpl.java:28]: Java doc, please.

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/Anno.java (hunk 1 above)
Comment (haifxu) [reply inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/Anno.java]: Done.

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/AnnoImpl.java (hunk 2 above)
Comment (haifxu) [reply inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/AnnoImpl.java]: Done.

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/NoAuth.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.manager.client.api;
+
+import org.apache.inlong.manager.pojo.user.UserRequest;
+
+/**
+ * No auth user login, register, etc.
```
Comment (healchow) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/NoAuth.java:23]: `No auth interface, such as user login, register, etc.`

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/NoAuthImpl.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import org.apache.inlong.manager.client.api.NoAuth;
+import org.apache.inlong.manager.client.api.ClientConfiguration;
+import org.apache.inlong.manager.client.api.inner.client.NoAuthClient;
+import org.apache.inlong.manager.client.api.inner.client.ClientFactory;
+import org.apache.inlong.manager.client.api.util.ClientUtils;
+import org.apache.inlong.manager.common.util.Preconditions;
+import org.apache.inlong.manager.pojo.user.UserRequest;
+
+/**
+ * No auth user implementation.
```
Comment (healchow) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/NoAuthImpl.java:29]: `No auth interface implementation` is enough.

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/NoAuth.java (hunk 3 above)
Comment (haifxu) [reply inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/NoAuth.java]: Done.

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/NoAuthImpl.java (hunk 4 above)
Comment (haifxu) [reply inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/NoAuthImpl.java]: Done.

**Review by healchow — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**